tests/integration/test_interface.py

```diff
@@ -140,26 +142,35 @@ def test_generate_dataset_and_col_noising(
         assert check_noised.loc[originally_missing_idx, col_name].isna().all()
 
         # Check for noising where applicable
-        shared_not_missing_idx = shared_idx.difference(originally_missing_idx)
+        to_compare_idx = shared_idx.difference(originally_missing_idx)
         if col.noise_types:
             assert (
-                check_original.loc[shared_not_missing_idx, col_name].values
-                != check_noised.loc[shared_not_missing_idx, col_name].values
+                check_original.loc[to_compare_idx, col_name].values
+                != check_noised.loc[to_compare_idx, col_name].values
             ).any()
+
+            noise_level = (
+                check_original.loc[to_compare_idx, col_name].values
+                != check_noised.loc[to_compare_idx, col_name].values
+            ).mean()
+
             # Check that the amount of noising seems reasonable
-            threshold = 1 - cell_probability ** len(col.noise_types)
-            # NOTE: The threshold is not perfect - it is very underconservative
-            #   because it does not account for token-level noising which could
-            #   result in no actual changes to a value despite being chosen
-            # TODO: generate more accurate thresholds
-            assert (
-                check_original.loc[shared_not_missing_idx, col_name].values
-                != check_noised.loc[shared_not_missing_idx, col_name].values
-            ).mean() <= threshold
-        else:
+            tmp_config = config[dataset.name][Keys.COLUMN_NOISE][col_name]
+            includes_token_noising = [
+                c for c in tmp_config if Keys.TOKEN_PROBABILITY in tmp_config[c].keys()
```

Comment:
Don't you also need the zipcode probabilities here?

Reply:
The zipcode is all the same now I thought?

Reply:
Rajan is saying that zipcode has a unique parameter set that behaves like token_probabability (`write_wrong_zipcode_digits.digit_probabilities`) and so should have been accounted for in this list comprehension. Turns out that the zipcode columns are also noised with `make_typos` which have token probability anyway so I just left it out. It would presumably break if we ever turned off `make_typos` and the rtol set to the lower 0.11 - or maybe it would pass because the default digit probabilities are higher than the default token probabiilties?